covid_graphs/covid_graphs/country_graph.py

```diff
@@ -118,30 +113,31 @@ def pick_xaxis_labels(object):
         )
         if graph_type != GraphType.Normal:
             maximal_y = max(
-                max(curve.maximal_y for curve in self.curves), max(self.cumulative_active)
+                max(curve.get_maximum()[1] for curve in self.curves),
+                max(self.cropped_cumulative_active),
             )
+            # Note: We silently assume `self.cropped_cumulative_active` does not contain zeros.
             layout.yaxis["range"] = [
-                math.log10(self.cumulative_active.min()) - 0.3,
+                math.log10(self.cropped_cumulative_active.min()) - 0.3,
                 math.log10(maximal_y) + 0.3,
             ]
 
         figure = Figure(layout=layout)
         for color, curve in zip(colors, self.curves):
             figure.add_trace(
                 Scatter(
-                    x=pick_xaxis_labels(curve),
-                    y=curve.y,
-                    text=curve.date_list,
+                    x=list(map(adjust_xlabel, curve.xs)),
```

Comment:
Previously, each curve was extended if it was shorter than the data of active cases. You need to take into account the length of `self.cropped_cumulative_active` when plotting it it.